rustls/src/key.rs

```diff
@@ -97,6 +99,20 @@ impl fmt::Debug for Certificate {
     }
 }
 
+/// wrapper around internal representation of a parsed certificate. This is used in order to avoid parsing twice when specifying custom verification
+#[allow(unreachable_pub)]
```

Comment:
I think we can tighten up this `allow` to only be in-effect when not using `dangerous_configuration`: 
```suggestion
#[cfg_attr(not(feature = "dangerous_configuration"), allow(unreachable_pub))]
```